api4/channel.go

```diff
@@ -1020,18 +1025,20 @@ func searchAllChannels(c *Context, w http.ResponseWriter, r *http.Request) {
 	includeDeleted, _ := strconv.ParseBool(r.URL.Query().Get("include_deleted"))
 	includeDeleted = includeDeleted || props.IncludeDeleted
 
+	// PolicyID is ignored since that may only be used through the /data_retention/policies endpoint
```

Comment:
We don't want to allow clients to use the `PolicyID` field via this API endpoint. The `model.ChannelSearch` struct is shared between both `/channels/search` and `/data_retention/policies/<policy_id>/channels/search`.

On a side note, we actually have this problem in a few other places, where API endpoints are accepting parameters which aren't officially documented because of the way we're sharing structs.